packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-alarm-action.ts

```diff
@@ -0,0 +1,49 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for CloudWatch alarm.
+ */
+export interface CloudWatchAlarmActionProps extends CommonActionProps {
+  /**
+   * The reason for the alarm change.
+   */
+  readonly stateReason: string;
+  /**
+   * The value of the alarm state.
+   */
+  readonly stateValue: string;
```

Comment:
1. This should not be string. It should be of the type [`AlarmState` from the CloudWatch library](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudwatch.AlarmState.html).
2. I'm not in love with this name. How about `alarmStateToSet` instead?

Reply:
Sounds good, using `cloudwatch.AlarmState` instead.